Source/WebKit/GPUProcess/graphics/WebGPU/RemoteAdapter.cpp

```diff
@@ -47,7 +47,9 @@ RemoteAdapter::RemoteAdapter(PAL::WebGPU::Adapter& adapter, WebGPU::ObjectHeap&
     , m_streamConnection(WTFMove(streamConnection))
     , m_identifier(identifier)
 {
-    m_streamConnection->startReceivingMessages(*this, Messages::RemoteAdapter::messageReceiverName(), m_identifier.toUInt64());
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
I'm not reviewing these very closely. Someone that knows GPUProcess should review these.

Reply:
@kkinnunen-apple is this a reasonable approach or should I do something else? These constructors get called on the WebGPU work queue but startReceivingMessages needs to be called from the main queue at the moment, right?

Reply:
On the surface level this looks incorrect.
If this is the intention, this class of changes look like they could perhaps be submitted separately.
Submitting to main runloop is done with callOnMainRunLoop, but this really doesn't look like the correct approach.

IIRC the WebGPU, like WebGL is using the "main" IPC::Connection as their out of stream connection.
If I understand correctly, you are running into an assertion? I think the solution is to start using the "dedicated connection", similar to what RemoteRenderingBackend uses. This change could be submitted separately.